components/iotserver-api/org.wso2.carbon.device.mgt.iot.common.api/src/main/java/org/wso2/carbon/device/mgt/iot/common/api/PolicyManagementService.java

```diff
@@ -0,0 +1,508 @@
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@GET
+	@Path("/start-task/{milliseconds}")
+	public boolean startTaskService(@PathParam("milliseconds") int monitoringFrequency) {
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			TaskScheduleService taskScheduleService = policyManagerService.getTaskScheduleService();
+			taskScheduleService.startTask(monitoringFrequency);
+			if (log.isDebugEnabled()) {
+				log.debug("Policy monitoring service started successfully.");
+			}
+			return true;
+		} catch (PolicyMonitoringTaskException e) {
+			String error = "Policy Management related exception.";
+			log.error(error, e);
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@GET
+	@Path("/update-task/{milliseconds}")
+	public boolean updateTaskService(@PathParam("milliseconds") int monitoringFrequency) {
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			TaskScheduleService taskScheduleService = policyManagerService.getTaskScheduleService();
+			taskScheduleService.updateTask(monitoringFrequency);
+			if (log.isDebugEnabled()) {
+				log.debug("Policy monitoring service updated successfully.");
+			}
+			return true;
+		} catch (PolicyMonitoringTaskException e) {
+			String error = "Policy Management related exception.";
+			log.error(error, e);
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@GET
+	@Path("/stop-task")
+	public boolean stopTaskService() {
+
+		try {
+			PolicyManagerService policyManagerService = getPolicyServiceProvider();
+			TaskScheduleService taskScheduleService = policyManagerService.getTaskScheduleService();
+			taskScheduleService.stopTask();
+			if (log.isDebugEnabled()) {
+				log.debug("Policy monitoring service stopped successfully.");
+			}
+			return true;
+		} catch (PolicyMonitoringTaskException e) {
+			String error = "Policy Management related exception.";
+			log.error(error, e);
+			return false;
+		} catch (DeviceManagementException e) {
+			String error = "Error occurred while invoking Policy Management Service.";
+			log.error(error, e);
+			response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+			return false;
+		} finally {
+			this.endTenantFlow();
+		}
+	}
+
+	@GET
+	@Path("/{type}/{id}")
+	public ComplianceData getComplianceDataOfDevice(@PathParam("id") String deviceId,
```

Comment:
why do not we simply return the javax.ws.rs.core.Response object here as well and make the return type consistent across all the APIs?